chain-signatures/node/src/protocol/presignature.rs

```diff
@@ -433,11 +482,6 @@ impl PresignatureManager {
                         participants = ?presig_participants.keys_vec(),
                         "running: the intersection of participants is less than the threshold"
                     );
-
-                    // Insert back the triples to be used later since this active set of
-                    // participants were not able to make use of these triples.
```

Comment:
I think instead of wasting the triples, we can actually change the order of ```
if let Some((triple0, triple1)) = triple_manager.take_two_mine().await
```
and 
```
if presig_participants.len() < self.threshold```
basically check participants.len() first, if that passes, then take_mine